app/scripts/lib/transaction/metrics.ts

```diff
@@ -351,6 +353,16 @@ export const handlePostTransactionBalanceUpdate = async (
   },
 ) => {
   if (getParticipateInMetrics() && transactionMeta.swapMetaData) {
+    // Prepare the transaction hash addition if feature flag is enabled
+    const shouldAddTransactionHash =
+      getRemoteFeatureFlags &&
```

Comment:
nit: there is a little shortcut: `getRemoteFeatureFlags?.()?.['transactions-tx-hash-in-analytics']`